fixtures/noop-132

```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env node
+'use strict';
+// Interleaving the output of stdout and stderr is unpredictable and can't be
+// expressed as a test. What I've tried so far to make it predictable:
+// - process.stdout._handle.setBlocking(true)
+// - process.stdout.once('drain', () => process.stderr.write('2'));
+// - process.stdout.write('1', () => process.stderr.write('2'));
+// - await delay() between calls; it works but with varying ms, based on load
```

Comment:
I think using delays of like 100ms might be good enough to make it somewhat predictable?

Reply:
👍

Reply:
Yes, the only way I found to get the test correct is to put a delay.
I'm testing right now on my machine and sometimes it passes the test with `3000ms` delay, but sometimes not. Which makes it unpredictable on other machines. I believe it's based on the current load on the system.
With `6000ms` delay, on my machine, it seems it always passes the test but slows down the whole testing. Is it enough as a solution? Have you any other idea on how it can be made predictable?

Reply:
Would the other solutions you're suggesting, e.g. using the `stream.write()` callback, work instead?

Reply:
Also I think it might be a good idea to do a more complex interleaving like stdout > stderr > stderr > stdout > stderr > stdout

Reply:
None of those worked for me. Only an high enough delay worked so far.
I've tried nesting callbacks on `stream.write`.
I've just re-tried some of what I've thought could work [here](https://gist.github.com/tomsotte/1e5fb674d94dba1cd7af1314af658c9d) (commenting/decommenting chunks of code to get the sequence).
Most of them returns on my console the correct sequence, interleaving stdout and stderr, '132', but when trying to test it they fail, returning `123`.
I may be missing something in all of this, but at the moment I've no other ideas to try.

Reply:
Did you try creating streams in the test, then passing those down to the executable file using the `stdout` and `stderr` options?

Reply:
Any other stream type passed down results to `null` for `stdout`, `stderr` and `all`. I tried with fs.createFileStream, net.Socket, 'inherit'... only the default 'pipe' works.
On ava it works, with a delay of `6000ms` on my machine before the last `stdout.write` call. That's the only way I can reproduce the result on `all` being interleaved.

Reply:
Other possible idea: using `test.serial()` to ensure no other tests are fired in parallel? This might be an issue due to parallelism. If it's not, this might still allow you to decrease the delay.

Reply:
Good find, `test.serial()` may be the answer. I've lowered to 1000ms the delay for `noop-132` and it seems to pass test runs consistently. Let's see if the CI agree as well.

Reply:
Seems like it works in CI, problem solved!